arrays/arrays.c

```diff
@@ -69,43 +75,81 @@ void resize_array(Array *arr) {
  *
  * Throw an error if the index is out of range.
  *****/
-char *arr_read(Array *arr, int index) {
-
+char *arr_read(Array *arr, int index)
+{
+  int errnum;
   // Throw an error if the index is greater than the current count
-
+  if (arr->count > index)
+  {
+    errnum = errno;
+    fprintf(stderr, "Value of errno: %d\n", errno);
+    perror("Error printed by perror");
+    fprintf(stderr, "Given index is higher than the size of the array");
+    return -1;
+  }
   // Otherwise, return the element at the given index
+  else
+  {
+    return arr->elements[index];
+  }
 }
 
-
 /*****
  * Insert an element to the array at the given index
  *****/
-void arr_insert(Array *arr, char *element, int index) {
-
+void arr_insert(Array *arr, char *element, int index)
+{
+  int errnum;
   // Throw an error if the index is greater than the current count
+  if (index > arr->count)
+  {
+    errnum = errno;
+    fprintf(stderr, "Value of errno: %d\n", errno);
+    perror("Error printed by perror");
+    fprintf(stderr, "Given index is higher than the size of the array");
+    return -1;
+  }
 
   // Resize the array if the number of elements is over capacity
+  if (arr->capacity == arr->count)
+  {
+    resize_array(arr);
+  }
 
   // Move every element after the insert index to the right one position
+  for (int i = arr->count; i >= index; i--)
```

Comment:
Good, this is how you want to do this to prevent duplication errors with the last element in the array.